torch/autograd/gradcheck.py

```diff
@@ -1167,6 +1188,18 @@ def _vec_from_tensor(x, generator, downcast_complex=False):
             .view(x_values.shape)
         values /= values.norm()
         vec = torch.sparse_coo_tensor(x._indices(), values, x.size())
+    elif _is_sparse_compressed_tensor(x):
+        if x.layout in {torch.sparse_csr, torch.sparse_bsr}:
+            compressed_indices, plain_indices = x.crow_indices(), x.col_indices()
+        else:
+            compressed_indices, plain_indices = x.ccol_indices(), x.row_indices()
+        x_values = x.values()
+        dtype = _to_real_dtype(x.dtype) if downcast_complex else x.dtype
+        values = torch.rand(x_values.numel(), generator=generator) \
+            .to(dtype=dtype, device=x.device) \
+            .view(x_values.shape)
+        values /= values.norm()
+        vec = torch.sparse_compressed_tensor(compressed_indices, plain_indices, values, x.size(), layout=x.layout)
```

Comment:
FYI this path is only used for fast gradcheck, so looks like it is currently not tested?

Reply:
The corresponding tests in `test_gradcheck_sparse_csr/csc_input` are disabled due to other issues that are OT for this PR.